samples/kotlin-android-app-destinations/src/main/java/com/segment/analytics/destinations/plugins/AmplitudeSession.kt

```diff
@@ -0,0 +1,95 @@
+package com.segment.analytics.destinations.plugins
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.lifecycle.ProcessLifecycleOwner
+import com.segment.analytics.*
+import com.segment.analytics.AndroidLogger.log
+import com.segment.analytics.platform.DestinationPlugin
+import com.segment.analytics.platform.plugins.LogType
+import com.segment.analytics.platform.plugins.log
+import com.segment.analytics.utilities.putIntegrations
+import java.time.Instant
+import java.util.*
+import kotlin.concurrent.schedule
+
+// A Destination plugin that adds session tracking to Amplitude cloud mode.
+class AmplitudeSession() : DestinationPlugin(), LifecycleObserver {
```

Comment:
i think we should use the `AndroidLifecycle` interface instead of `LifecycleObserver`